src/lib/dapp/DAppSigner.ts

```diff
@@ -32,18 +33,53 @@ import {
 } from '@hashgraph/sdk'
 import type { ISignClient } from '@walletconnect/types'
 
-import { ledgerIdToCAIPChainId } from '../shared'
+import {
+  ExecuteTransactionResult,
+  HederaJsonRpcMethod,
+  SignAndExecuteQueryResult,
+  SignAndExecuteTransactionResult,
+  SignMessageResult,
+  SignTransactionResult,
+  base64StringToSignatureMap,
+  ledgerIdToCAIPChainId,
+  transactionBodyToBase64String,
+  transactionToBase64String,
+  transactionToTransactionBody,
+} from '../shared'
 
 export class DAppSigner implements Signer {
+  static signClient: ISignClient
+  public signerAccountId: string
+
+  /**
+   * These are the node accounts that will be utilized for signing transactions.
+   */
+  public nodesAccountIds: AccountId[] = []
+
   constructor(
     private readonly accountId: AccountId,
-    private readonly signClient: ISignClient,
     public readonly topic: string,
     private readonly ledgerId: LedgerId = LedgerId.MAINNET,
-  ) {}
+  ) {
+    this.signerAccountId = `${ledgerIdToCAIPChainId(ledgerId)}:${accountId.toString()}`
+    this.nodesAccountIds = [AccountId.fromString('0.0.3')]
```

Comment:
`this.nodesAccountIds = [AccountId.fromString('0.0.3')]` is a single point of failure on consensus node 3. Consider creating a hashgraph/sdk client and grabbing node ids from it's _network property. See this Signer implementation from the hashgraph/sdk https://github.com/hashgraph/hedera-sdk-js/blob/main/src/Wallet.js#L272

Reply:
With the sdk client, you don't ensure you use the same nodes as the wallet you are using. So I think the best option is to set manually the available nodes after requesting them to the wallet. Anyway, I think here we have a good point to improve. I suggest creating a new PR to address the issue.

Reply:
> With the sdk client, you don't ensure you use the same nodes as the wallet you are using.
I don't think this should be a requirement for walletconnect. The `nodeAccountIds` on a transaction can differ from the nodes returned by a wallet, as long as they are nodes for the same network.

Reply:
@franfernandez20 we absolutely cannot be hardcoding node id's like this, if node 3 goes down all transactions using this node will fail

Reply:
To set the list, you can use the setNodes function. It is essential to ensure that the nodes in the client and wallet are aligned. Although we need to restore the Transaction.toBytes() approach to retain more nodes, we now have to use this method with only one node. 

Anyway, what do you propose?